### PR TITLE
Use the DumbAwareBGTAction interface to silence the update thread warning.

### LINKED ISSUE #1327: bug: com.intellij.diagnostic.PluginException: `ActionUpdateThread.OLD_EDT` is deprecated and going to be removed soon. 'com.source...
### Cody Version

5.5.5-nightly

### IDE Information

Golang 2024.1

### Describe the bug

n/a

### Expected behavior

n/a

### Additional context

Stacktrace:
```text
com.intellij.diagnostic.PluginException: `ActionUpdateThread.OLD_EDT` is deprecated and going to be removed soon. 'com.sourcegraph.website.OpenRevisionAction' must override `getActionUpdateThread` and chose EDT or BGT. See ActionUpdateThread javadoc. [Plugin: com.sourcegraph.jetbrains]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:90)
	at com.intellij.diagnostic.PluginException.reportDeprecatedUsage(PluginException.java:125)
	at com.intellij.openapi.actionSystem.ActionUpdateThreadAware.getActionUpdateThread(ActionUpdateThreadAware.java:21)
	at com.intellij.openapi.actionSystem.AnAction.getActionUpdateThread(AnAction.java:199)

```

### PR DESCRIPTION
**Pull request by odisseus**
This should fix #1327.

## Test plan
I have clicked around a little bit in IntelliJ IDEA 2024.1.1 RC, and I haven't seen any OLD_EDT warnings in the log.

### CODE REVIEW

**Review by pkukielka — APPROVED** [review]
LGTM

File: src/main/java/com/sourcegraph/website/OpenRevisionAction.java [hunk 1]
```diff
@@ -22,8 +22,7 @@
 import org.jetbrains.annotations.NotNull;
 
 /** JetBrains IDE action to open a selected revision in Sourcegraph. */
-@SuppressWarnings("MissingActionUpdateThread")
-public class OpenRevisionAction extends DumbAwareAction {
+public class OpenRevisionAction extends DumbAwareBGTAction {
```
Comment (pkukielka) [inline src/main/java/com/sourcegraph/website/OpenRevisionAction.java:25]: Unfortunately there are more actions to fix lie this.

Comment (odisseus) [reply src/main/java/com/sourcegraph/website/OpenRevisionAction.java:25]: Updated

File: src/main/java/com/sourcegraph/cody/PromptPanel.kt [hunk 2]
```diff
@@ -115,11 +114,7 @@ class PromptPanel(project: Project, private val chatSession: ChatSession) : JLay
           }
         })
     for (shortcut in listOf(ENTER, UP, DOWN, TAB)) { // key listeners
-      object : DumbAwareAction() {
-            override fun actionPerformed(e: AnActionEvent) {
-              didUseShortcut(shortcut)
-            }
-          }
+      SimpleDumbAwareBGTAction { didUseShortcut(shortcut) }
```
Comment (pkukielka) [inline src/main/java/com/sourcegraph/cody/PromptPanel.kt:117]: 👍

File: src/main/kotlin/com/sourcegraph/cody/context/ui/RemoteRepoPopupController.kt [hunk 3]
```diff
@@ -124,7 +124,7 @@ class RemoteRepoPopupController(val project: Project) {
             .createPopup()
 
     val okAction =
-        object : DumbAwareAction() {
+        object : DumbAwareBGTAction() {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/context/ui/RemoteRepoPopupController.kt:127]: `SimpleDumbAwareBGTAction`? :)

File: src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt [hunk 4]
```diff
@@ -19,7 +20,7 @@ private constructor(title: String, content: String, shouldShowUpgradeOption: Boo
   init {
     icon = Icons.CodyLogo
     val learnMoreAction: AnAction =
-        object : DumbAwareAction("Learn more") {
+        object : DumbAwareBGTAction("Learn more") {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt]: `SimpleDumbAwareBGTAction`

File: src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt [hunk 5]
```diff
@@ -31,16 +32,11 @@ private constructor(title: String, content: String, shouldShowUpgradeOption: Boo
...
-          override fun actionPerformed(anActionEvent: AnActionEvent) {
-            hideBalloon()
-          }
-        }
+    val dismissAction: AnAction = SimpleDumbAwareBGTAction("Dismiss") { hideBalloon() }
 
     if (shouldShowUpgradeOption) {
       val upgradeAction: AnAction =
-          object : DumbAwareAction("Upgrade") {
+          object : DumbAwareBGTAction("Upgrade") {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt]: `SimpleDumbAwareBGTAction`

File: src/main/kotlin/com/sourcegraph/common/ui/DumbAwareBGTAction.kt [hunk 6]
```diff
@@ -21,3 +22,12 @@ abstract class DumbAwareBGTAction : DumbAwareAction {
       icon: Icon?
   ) : super(text, description, icon)
 }
+
+class SimpleDumbAwareBGTAction(
+    text: @NlsActions.ActionText String? = null,
+    private val action: () -> Unit
+) : DumbAwareAction(text) {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/common/ui/DumbAwareBGTAction.kt]: How is that supposed to work? Shouldn't it inherit from `DumbAwareBGTAction`?

Comment (odisseus) [reply src/main/kotlin/com/sourcegraph/common/ui/DumbAwareBGTAction.kt]: Good catch!

File: src/main/kotlin/com/sourcegraph/cody/context/ui/RemoteRepoPopupController.kt (hunk 3 above)
Comment (odisseus) [reply src/main/kotlin/com/sourcegraph/cody/context/ui/RemoteRepoPopupController.kt:127]: The `SimpleDumbAwareBGTAction` doesn't give the lambda access to the action event class. In its current form, it isn't applicable when the event is actually used, though I may consider refactoring it to remove this restriction. 

**P.S.** This action also uses `AnAction#unregisterCustomShortcutSet`, so the simple wrapper is definitely not applicable here.

File: src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt (hunk 4 above)
Comment (odisseus) [reply src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt]: Refactored

File: src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt (hunk 5 above)
Comment (odisseus) [reply src/main/kotlin/com/sourcegraph/common/UpgradeToCodyProNotification.kt]: Refactored

**Pull request merged**